gtn/functions.cpp

```diff
@@ -110,4 +110,27 @@ DISPATCH1(forwardScore)
 DISPATCH1(viterbiScore)
 DISPATCH1(viterbiPath)
 
+Graph ctcLoss(
```

Comment:
I think this should live in a new file called `criteria` and maybe even in a sub namespace like `gtn::criteria::ctc`. What do you think about that?

Reply:
sure... that sounds good